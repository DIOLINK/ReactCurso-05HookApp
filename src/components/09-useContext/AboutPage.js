import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const AboutPage = () => {
  const { user, setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };

  return (
    <div>
      <h1>About</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
      <button className="btn btn-warning" onClick={handleClick}>
        Logout
      </button>
    </div>
  );
};

export default AboutPage;
