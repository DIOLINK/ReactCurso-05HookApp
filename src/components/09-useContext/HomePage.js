import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const HomePage = () => {
  const { user } = useContext(UserContext);

  return (
    <div>
      <h1>Home</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};

export default HomePage;
