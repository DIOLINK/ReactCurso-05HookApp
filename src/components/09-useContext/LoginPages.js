import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginPage = () => {
  const { setUser } = useContext(UserContext);
  const user = { id: new Date().getTime(), name: 'John Doe' };
  return (
    <div>
      <h1>Login</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(user)}>
        Login
      </button>
    </div>
  );
};

export default LoginPage;
