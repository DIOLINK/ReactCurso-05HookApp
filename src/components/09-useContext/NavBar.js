import React from 'react';
import { Link, NavLink } from 'react-router-dom';
const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <Link className="navbar-brand" to="/">
          useContext
        </Link>
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink
                exact="true"
                className={({ isActive }) =>
                  `${isActive ? 'nav-link active' : 'nav-link'}`
                }
                aria-current="page"
                to="/"
              >
                Home
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                exact="true"
                className={({ isActive }) =>
                  `${isActive ? 'nav-link active' : 'nav-link'}`
                }
                to="/about"
              >
                About
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                exact="true"
                className={({ isActive }) =>
                  `${isActive ? 'nav-link active' : 'nav-link'}`
                }
                to="/login"
              >
                Login
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
