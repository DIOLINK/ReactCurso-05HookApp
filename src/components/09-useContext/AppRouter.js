import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';

import NavBar from './NavBar';
import AboutPage from './AboutPage';
import HomePage from './HomePage';
import LoginPage from './LoginPages';
const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route exact="true" path="/" element={<HomePage />} />
            <Route exact="true" path="/about" element={<AboutPage />} />
            <Route exact="true" path="/login" element={<LoginPage />} />
            <Route path="*" element={<HomePage />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
};

export default AppRouter;
