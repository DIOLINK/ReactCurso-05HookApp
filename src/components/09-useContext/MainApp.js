import React, { useState } from 'react';
import AppRouter from './AppRouter';
import { UserContext } from './UserContext';
// import './styles.css';
const MainApp = () => {
  const [user, setUser] = useState({});
  return (
    <UserContext.Provider value={{ user, setUser }}>
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
