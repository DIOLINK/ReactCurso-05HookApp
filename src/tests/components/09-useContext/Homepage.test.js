import { mount } from 'enzyme';
import HomePage from '../../../components/09-useContext/HomePage';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Prubas en <HomePage />', () => {
  const user = { name: 'Roger', email: 'Roger@mail.com' };
  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomePage />
    </UserContext.Provider>
  );

  it('Debe de mostrar el componente <HomePage /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
