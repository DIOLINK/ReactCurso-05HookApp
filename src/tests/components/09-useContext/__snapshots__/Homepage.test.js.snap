// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas en <HomePage /> Debe de mostrar el componente <HomePage /> correctamente 1`] = `
<div>
  <h1>
    Home
  </h1>
  <hr />
  <pre>
    {
   "name": "Roger",
   "email": "Roger@mail.com"
}
  </pre>
</div>
`;
